packages/mobile/src/backup/BackupComplete.tsx

```diff
@@ -2,65 +2,48 @@ import Checkmark from '@celo/react-components/icons/Checkmark'
 import fontStyles from '@celo/react-components/styles/fonts'
 import { StackScreenProps } from '@react-navigation/stack'
 import * as React from 'react'
-import { WithTranslation } from 'react-i18next'
+import { useTranslation } from 'react-i18next'
 import { StyleSheet, Text, View } from 'react-native'
 import { SafeAreaView } from 'react-native-safe-area-context'
-import { connect } from 'react-redux'
 import { OnboardingEvents } from 'src/analytics/Events'
 import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
-import { withTranslation } from 'src/i18n'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
-import { RootState } from 'src/redux/reducers'
-
-interface StateProps {
-  backupCompleted: boolean
-}
-
-type Props = StateProps & WithTranslation & StackScreenProps<StackParamList, Screens.BackupComplete>
-
-const mapStateToProps = (state: RootState): StateProps => {
-  return {
-    backupCompleted: state.account.backupCompleted,
-  }
-}
+import useSelector from 'src/redux/useSelector'
 
 /**
  * Component shown to the user upon completion of the Recovery Phrase setup flow. Informs the user that
  * they've successfully completed the backup process and automatically returns them to where they
  * came from.
  */
-class BackupComplete extends React.Component<Props> {
-  static navigationOptions = { header: null }
 
-  componentDidMount() {
-    // Show success check for a while before leaving screen
-    const { backupCompleted } = this.props
-    setTimeout(() => {
-      const navigatedFromSettings = this.props.route.params?.navigatedFromSettings ?? false
-      if (navigatedFromSettings) {
-        navigate(Screens.Settings, { promptConfirmRemovalModal: true })
-      } else if (backupCompleted) {
-        ValoraAnalytics.track(OnboardingEvents.backup_complete)
-        navigate(Screens.BackupIntroduction)
-      } else {
-        throw new Error('Backup complete screen should not be reachable without completing backup')
-      }
-    }, 2000)
-  }
+type Props = StackScreenProps<StackParamList, Screens.BackupComplete>
+
+function BackupComplete({ route }: Props) {
+  const navigatedFromSettings = route.params?.navigatedFromSettings ?? false
+  const backupCompleted = useSelector((state) => state.account.backupCompleted)
+  const { t } = useTranslation()
+
+  setTimeout(() => {
```

Comment:
This is setting the timeout on every render, but we only want it to happen on mount looks like. To achieve that we need a `useEffect` hook, like so:
```
useEffect(() => {
  setTimeout(...)
}, [])
```
notice the empty array as a second parameter, that array is a list of dependencies, whenever one of those changes the function in the first parameter is called (it's also called the first time). Since we're passing an empty array nothing can change so the whole thing is only ran once.